### PR TITLE
Wp rise article

### PR DESCRIPTION
**Pull request by willpoulett**
# Description

❓**What**: New blog on RISE tool.

🧠**Why?**: New content

👨‍💻**How?**: New article added, plus corresponding images.

# Checklist:
Have checked for the following:
- [ ] The website still builds correctly, and you can view it using `mkdocs serve`.
- [ ] There are no new "warnings" from mkdocs
- [ ] Does your page follow the [page template](https://nhsdigital.github.io/rap-community-of-practice/example_RAP_CoP_page/) (or [here in Markdown](https://github.com/NHSDigital/rap-community-of-practice/blob/main/docs/example_RAP_CoP_page.md))? (**need to make a new one specific to NHSE Data Science**)
- [ ] Spelling errors
- [ ] Consistent capitalization
- [ ] Consistent numbers
- [ ] Material features incorrectly implemented: search for code blocks and markers (e.g. !!!).
- [ ] Code snippets don't work
- [ ] Images not working
- [ ] Links not working

## Where it was tested

- Github Codespaces - 2-core, 4GB RAM, 32GB hard drive
- devcontainer.json describes further settings

### CODE REVIEW

File: docs/articles/posts/20241203_RISE_article.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+<figcaption>Figure 8: Two images that appear very similar but were classified differently by the model. This example was found by a user using the tool.</figcaption>
+</figure>
+
+When labels were included, users were similarly successful, finding many of the same examples of those mentioned above.
+
+### Would you want to generate any additional images for testing?
+
+When labels were not included, users wanted to see more kittens, puppies and cubs, animals with their eyes closed, animals in action and more wildlife images. One user specified it would be useful to find images that are close to identical but with key features changed – this might be easier to implement now due to the recent release of '[Add it](https://research.nvidia.com/labs/par/addit/)'.  With labels, users wanted to look further into domestic cats.
+
+### Did you find this tool useful and are there any improvements you’d like to see?
```
Comment (amaiaita) [inline docs/articles/posts/20241203_RISE_article.md:226]: do you need to give this much detail on the user feedback in a blog? i think just a couple of bullet points of feedback and what your actions from it are going to be would be more appropriate for this format

File: docs/articles/posts/20241203_RISE_article.md [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+
+I did notice that within the feedback there was some confusion regarding whether some behaviour was part of the tool or part of the AI classifier. One user suggested that the tool should have another class called 'other'. Whilst this is a good insight, it is in fact the model which would need another class. Additionally, some users mixed up terms such as efficiency and accuracy.
+
+This highlights the need for thorough staff training when using this tool, and many suggestions regarding the tools usability should be acted upon before the next iteration of this work.
+
+Additionally, none of the feedback referenced any edge-case (hybrid animal) images, instead pointing out occurrences when easily identifiable animals were performing certain behaviours or contained certain features. This is interesting, and may imply that the edge-case images we used were not found to be very helpful by the assurers. Alternatively, our guess of what an edge-case image was might have been completely wrong – the model interpreted images in a completely different way to human users.
+
+## Conclusion and Future Work
```
Comment (amaiaita) [inline docs/articles/posts/20241203_RISE_article.md:266]: i like this section a lot, but maybe with it being a blog you could name is something more jazzy like "How successful were we?" and "What's next?"

File: docs/articles/posts/20241203_RISE_article.md [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+description: >
+    This article discusses a method for evaluating AI classifiers using a new tool called RISE. 
+---
+
+> We have built a proof-of-concept tool which will help assurers, data scientists and clinicians to evaluate AI classifiers.
+ 
+> We call this the RISE tool, it utilises LLM's, AI Image Generators and an interactive plot to allow users to easily evaluate image classifiers.
+
+> We carried out careful experimentation to ensure its effectiveness, and plan to continue this research in the future.
+
```
[the remark this answers is not in the document]
Comment (willpoulett) [reply docs/articles/posts/20241203_RISE_article.md:24]: Done :)

File: docs/articles/posts/20241203_RISE_article.md [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+3. We wanted high quality documentation.
+4. We wanted the dataset to have some similarities to medical datasets. We see little variation between images in medical datasets (consider chest X-rays), and therefore wanted to replicate this with our dataset. As the Animals Faces Dataset only contained the face of animals in each image, this resulted in similar images across the dataset.
+5. We wanted to easily generate synthetic images. AI image generators are very good at generating images of pets and wildlife. Whilst they may also be good at generating medical images, without experts we cannot be sure of their quality.
+
+<figure class = "inline end" markdown>
+![3 Example Images from the Animal Faces Dataset](../../images/RISE_tool/Figure_2.png)
+<figcaption>Figure 2: Images from the Animal Faces Dataset. </figcaption>
+</figure>
+
+It goes without saying that we intend to use this tool on medical datasets in the future, with guidance from clinicians as to how realistic and useful AI generated medical images are. If you want to read ahead, our exact plans on future research can be found at the end of this article.
```
[the remark this answers is not in the document]
Comment (willpoulett) [reply docs/articles/posts/20241203_RISE_article.md:86]: woah I didn't know you could do that

File: docs/articles/posts/20241203_RISE_article.md [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+
+An assurance college was given the initial evaluation results alongside a description of the dataset and generated an initial list of 14 scenarios for us to test.  These included 'domestic dogs that look like wild dogs' and 'multiple animals in one picture'. We asked Llama 3.1 8B with a temperature of 0.7 to generate an additional list of scenarios, and it did so generating a list of 44 new scenarios. We then asked it again to consider it's previous risks and generate some more, this time adding 15 new scenarios. Whenever we used an LLM we followed good prompt guidance, this included asking the model to adopt a persona, asking the model if it missed anything on previous passes and giving examples.
```
[the remark this answers is not in the document]
Comment (willpoulett) [reply docs/articles/posts/20241203_RISE_article.md:104]: good spot, added a link too

File: docs/articles/posts/20241203_RISE_article.md [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+Some example images are shown in Figure 4, including 'edge-case' images.
+
+<figure markdown>
+![Diagram showing three AI generated images](../../images/RISE_tool/Figure_4.png)
+<figcaption> Figure 4: A sample of AI generated images in our dataset. </figcaption>
+</figure>
+
+### Edge-Case Examples
+
+We wanted to pay particular attention to edge-case image examples, as this mirrors boundary analysis in traditional software testing. Some of our scenarios already included edge-case examples. This included 'Crossbreed or hybrid animals', which we hoped would lie closer to the model's decision boundary and would help us identify where the model changes its decision, and which features in an image correspond to this.
```
[the remark this answers is not in the document]
Comment (willpoulett) [reply docs/articles/posts/20241203_RISE_article.md:138]: Added a single sentence, should add a bit more context without too much jargon.

File: docs/articles/posts/20241203_RISE_article.md [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+
+Additionally, we used GPT4o and DallE3 to generate 40 more  image prompts for edge case scenarios – in particular hybrid animals. These were animals that had features from multiple classes, and to a human were hard to classify. In a medical dataset, this may a certain disease with symptoms like an alternative disease.
```
[the remark this answers is not in the document]
Comment (willpoulett) [reply docs/articles/posts/20241203_RISE_article.md:140]: impressive spot

File: docs/articles/posts/20241203_RISE_article.md [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+
+If an assurer were to inspect these clusters, they may find occurrences where a certain group of similar images are misclassified. They might even have similar features or themes which can then be used to identify risks within the model. With dogs and cats, perhaps a cluster of images of cats holding tennis balls are all misclassified as dogs. In a clinical chest X-ray dataset, perhaps a chest X-ray with a broken rib is instead classified as having a tumour. 
+
+### Tool Design
+
+We used Bokeh to create our tool that can be accessed within a Jupyter Notebook. The tool was essentially an interactive scatter plot, where you could use a slider to navigate between different dimension reduction techniques.
```
[the remark this answers is not in the document]
Comment (willpoulett) [reply docs/articles/posts/20241203_RISE_article.md:198]: The rest of the article is past tense so I'd rather be consistent.

File: docs/articles/posts/20241203_RISE_article.md [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+
+There are plentiful ways to improve this section of the experiment for future studies. If we are to move on to a medical dataset and let clinicians use the tool, we may wish to time how long it takes them to label each image and therefore remove certain votes if not enough time was taken. We may also want a back button or an 'I don't know' button, alongside using a larger cohort of labellers to try and get statistically significant results. 
+
+## Interactive Tool
+
+Let's finally talk about the interactive tool. We had synthetic images, we had labels, all that remained was to create a clear way of plotting and interacting with them. 
+
+### Dimension Reduction
+
+To create our scatter plots, we needed a way to turn model predictions into a set of two-dimensional coordinates. This was achieved using hidden layer activations and dimension reduction techniques.
```
[the remark this answers is not in the document]
Comment (willpoulett) [reply docs/articles/posts/20241203_RISE_article.md:186]: Reworded the paragraphs after a tiny bit, but these concepts are explained there.

File: docs/articles/posts/20241203_RISE_article.md [hunk 10]
```diff
@@ -0,0 +1,301 @@
...
+
+| Labels              | Domestic Cat | Domestic Dog | Wildlife |
+|---------------------|--------------|--------------|----------|
+| Predictions shown   | 49           | 38           | 61       |
+| Predictions not shown | 49           | 36           | 63       |
+| All labels          | 53           | 37           | 58       |
+
+The above table shows the different label counts when users were shown predictions, were not shown predictions and when considering both scenarios. Whilst there are differences, they are small. We knew that some edge case images were hard to classify and received equal numbers of votes for multiple classes. We also knew that some users made mistakes whilst labelling these images with the label tool. This might have explained some of the small differences in labelling - it does not seem like the prediction being shown had a significant impact.
```
[the remark this answers is not in the document]
Comment (willpoulett) [reply docs/articles/posts/20241203_RISE_article.md:172]: tried to reword a bit

File: docs/articles/posts/20241203_RISE_article.md (hunk 1 above)
Comment (willpoulett) [reply docs/articles/posts/20241203_RISE_article.md:226]: cut down significantly

File: docs/articles/posts/20241203_RISE_article.md [hunk 11]
```diff
@@ -0,0 +1,301 @@
+---
+title: The RISE Tool – An Easy Way for Testers and Assurers to Evaluate AI Classifiers
+authors: [willpoulett]
+date: 2024-12-03
```
[the remark this answers is not in the document]
Comment (willpoulett) [reply docs/articles/posts/20241203_RISE_article.md:4]: updated to today :)

File: docs/articles/posts/20241203_RISE_article.md [hunk 12]
```diff
@@ -0,0 +1,301 @@
...
+
+### Model Setup
+
+Once our dataset was selected, it was time to train a model. Our model was trained using transfer learning on top of the [EfficientNetV2S](https://arxiv.org/abs/2104.00298) model, with ImageNet weights. The model performed exceptionally well on a test dataset containing 1467 images (493, 491 and 483 images for cats, dogs and wildlife respectively) with 99.8% accuracy. The precision for the cat and dog class was perfect. The only incorrect classifications were three images, all predicted as a wild animal when their label was either a cat or a dog. 
+
+The three images that were incorrectly classified are shown in Figure 3. The left-most image stands out most due to a possible instance of label noise. The image is likely a clouded leopard – wildlife, yet has a ground truth label of a cat. If we have identified possible label noise in the test dataset, we can assume there are probably instances of it in the training dataset.
```
[the remark this answers is not in the document]
Comment (willpoulett) [reply docs/articles/posts/20241203_RISE_article.md:88]: Think you were looking at figure 2 not 3! Made some changes to the paragraph.

**Review by amaiaita — APPROVED** [review]

**Pull request merged**